rpc/client.go

```diff
@@ -191,6 +193,31 @@ func (c *Client) Healthy() <-chan struct{} {
 	return c.healthy.Load().(chan struct{})
 }
 
+// WaitHealthy blocks until the channel is healthy or the HealthWait time has
+// expired.
+func (c *Client) WaitHealthy() bool {
```

Comment:
This probably deserves a test in this package. Consider this PR a work in progress.